client/worker.go

```diff
@@ -189,6 +192,17 @@ func (w *worker) prepare(ctx context.Context, batches []*jaegerpb.Batch, spansCo
 		return nil, err
 	}
 
+	if w.disableCompression {
+		return &sendRequest{
+			message: encoded,
+			batches: int64(len(batches)),
+			spans:   int64(spansCount),
```

Comment:
Nit: duplicate code (maybe move these 2 fields initialization before `if`).

Reply:
but in case of disableCompression = false and errors down the line, they are not being used. Do you still think better to move them out?